modules/nextflow/src/test/groovy/nextflow/scm/RepositoryProviderTest.groovy

```diff
@@ -108,4 +108,15 @@ class RepositoryProviderTest extends Specification {
         and:
         1 * conn.setRequestProperty('Authorization', "Basic ${'foo:bar'.bytes.encodeBase64()}")
     }
+
+    def 'should have public revision property' () {
+        given:
+        def provider = Spy(RepositoryProvider)
+        when:
+        provider.revision = 'branch_or_tag'
+        then:
+        provider.revision == 'branch_or_tag'
+        provider.hasProperty('revision') ? true : false
+    }
```

Comment:
Do we need a unit test for this? I don't think the test will actually test anything because Spock can access protected members on a spy anyway

Reply:
The property getter (i.e. `provider.revision == 'branch_or_tag'`), definitely raises a MissingMethodException exception if the field is protected. Although tbh I wasn't expecting that as the test is in the same package, so should have access. The `hasProperty` test was my attempt to assert that is is actually property and hence public.

This is however moot given Paolo's comments below which I will address. And this test will go away.

Reply:
Another option here would be to have test in another package to assert the public API. I don't currently see this pattern in the repo though, so will take you guidance on whether this is required of not.